Diogenes

jwmerrill656 opened the pull request:
build and learn diogenes basics

pjheslin committed:
Serve images from static internet site for DiogenesWeb

pjheslin committed:
Lowercase beta code generated from utf8 for Perseus compatibility.

Fixes bug with parsing capitalised vowels in non-proper-nouns at the
start of sentences, where we have to try lower case.

pjheslin committed:
Add port to Dockerfile

pjheslin committed:
File for packaging as a single container for AWS.

pjheslin committed:
Fix for Logeion link.

pjheslin committed:
Remove OLD and TLL links from DiogenesWeb

pjheslin committed:
Add --init Docker flag on AWS

pjheslin committed:
Less dict padding for DWeb

pjheslin committed:
Prefork more children for morph server

pjheslin committed:
Throw error when morph server does not get user type param

pjheslin committed:
Deploy morph server to eb

pjheslin commented:
Nick is no longer maintaining his branch of Diogenes.  Please use pjheslin/diogenes instead, which is where development is happening.

pjheslin committed:
Remove single quotation marks when parsing Latin (not Greek)

Fix problem introduced when fixing parsing of Greek elided forms in commit 044a6cd5fa5cdb14c82be2faece3b11bb3963cdd

pjheslin committed:
Fix treatment of y with acute accent in Latin texts.

pjheslin committed:
Fix description of location of menu item

pjheslin committed:
Turn off debugging in morphology server by default

pjheslin committed:
Update Unicode Greek punctuation in line with TLG recommendations

pjheslin committed:
to-dos

pjheslin committed:
Fix double conversion to utf8

Not sure how this started to happen.

pjheslin committed:
Highlight word in text that has been clicked.

pjheslin committed:
Added current dir to @INC for all scripts.

It has been removed for security reasons from the Perl core.

pjheslin committed:
Fix Readme to point to icon packages that need to be installed first

pjheslin committed:
Take Lewis-Short lexicon from Logeion github

Refactor makefiles to separate Perseus and Logeion.

pjheslin committed:
Fix crash when exporting corrupted input files as XML.

Many people have a corrupted version of Caesar, processing which made
the whole export process die.

pjheslin committed:
Fix Windows bug where subcorpus persisted in new searches.

Even when deselected.  Manifested when using the non-forking server
that does not die at the end of serving client.

pjheslin committed:
Fix building Windows app, which broke in OS X Catalina.

Even 32-bit emulated Windows apps are prevented from running under
Catalina, and there is no 32-bit version of Inno Setup, so we have to
use Docker for now instead of Wine.

pjheslin committed:
Clarify text.

pjheslin committed:
When upgrading app, check version and clear HTTP cache.

Unless we do this, JS and CSS from the previous version will stick
around despite the new install, causing tough-to-find bugs.

pjheslin committed:
Make sorting of authors stable when names are identical.

In these cases, as with Aescylus father and son, we want to keep the
numbers in there rather than removing them.

pjheslin committed:
Version 4.3

pjheslin committed:
Clarify Readme

pjheslin committed:
Fix warning in old CGI.pm

pjheslin committed:
Minor changes to makefiles

pjheslin committed:
Fixes to morph server to use it with DiognesWeb lookup popups

pjheslin committed:
Support for eb deploy command for morph-server

pjheslin committed:
In fullscreen mode, prevent fullscreen for find mini-window.

pjheslin committed:
Update Changes

pjheslin committed:
Fix strange windows bug that causes hanging when parsing

In some but not all Windows computers (but it does not depend on
version number), clicking on a word to parse it causes a lengthy hang
the first time it is done.  Not sure why, but it was slurping the
whole lexicon file and trying to parse it.  Somehow $/ was being
unset. Removing local fixes the bug; not sure why.

pjheslin committed:
Tidying indentation.

pjheslin committed:
Fix inconsistent printing of bibliographical info.

On Windows, the server does not die, so the TLG bibliography file
stays in memory as $bibliography, but the object is new, so that fact
was not being flagged.

pjheslin committed:
Fix duplicate filter names on Windows.

Because of persistent state across requests.

pjheslin committed:
4.4 prerelease

pjheslin committed:
Fixed bug where new windows did not use the adjusted window size.

pjheslin committed:
Restore inadvertently removed word index search on subset of TLG.

pjheslin committed:
Better error handling for foregoing.

pjheslin committed:
... and update to-do.

pjheslin committed:
Add facility for user-defined CSS.

pjheslin committed:
Modify tip on Greek text entry to include Beta code.

pjheslin committed:
Fixed broken morphological search.

pjheslin committed:
Update .gitignore

pjheslin committed:
Modify code to permit two user-configurable css files.

pjheslin committed:
Added font select menu item.

pjheslin committed:
Add Loading... message for font list: esp. slow on Windows.

pjheslin committed:
Version 4.4

pjheslin committed:
Fix where TLL corrigenda items were overwriting correct bookmarks.

pjheslin committed:
Fix makefile

pjheslin committed:
Final change for 4.4

pjheslin committed:
Fix parsing of ὄνος.

pjheslin committed:
Don't automatically rebuild installers when releasing.

pjheslin committed:
Style Logeion link properly.

pjheslin committed:
Fix parsing of Greek words with Unicode marks of elision.

pjheslin committed:
Add Diogenes version number to parses from DiogenesWeb.

pjheslin committed:
Fix warning.

pjheslin committed:
Fix error looking for OLD entries for words with accents.

pjheslin committed:
Remove debugging statements.

pjheslin committed:
Fix TLL PDF downloads after BAdW website was reorganized.

pjheslin committed:
Fix same issue for TLL as for OLD words with accents.

pjheslin committed:
Improve parsing of phrases in TLL bookmarks.

pjheslin committed:
Fixed bug in morphological search when Perseus lemma has a comma in.

pjheslin committed:
Version 4.5

pjheslin committed:
Use orth_orig for LSJ headwords in Logeion, and fix entities in them

pjheslin committed:
Remove debugging comment

pjheslin committed:
Clean up debug statement.

luxzeitlos committed:
change shebang from /usr/bin/perl to /usr/bin/env

Generally using /usr/bin/perl is not a good shebang
because perl does not always live at /usr/bin/perl.
This is, for example, true for NixOS where multiple
versions of perl may be installed side by side in
multiple shells and for multiple users.
the correct way to lookup a program in the shebang
is /usr/bin/env perl.
Because here the shebang was /usr/bin/perl -w we
must use the -S switch on /usr/bin/env to pass
arguments to perl.
So the correct shebang is /usr/bin/env -S perl -w.

This applies to all kind of scripts. So also
/usr/bin/bash is not a good shebang. It should be
/usr/bin/env bash.

pjheslin committed:
Provide fix for when windows are displayed off-screen.

pjheslin committed:
Refactor find-in-page to fix bug of multiple find windows

(cherry picked from commit 3ca5b4c498d1ce687a959c16dd1eef09327f256e)

pjheslin committed:
TODO update

pjheslin committed:
Make sure to pick up GUI config when running from Linux command-line

pjheslin committed:
Disable browse forward button after hitting end of file.

pjheslin committed:
Fix browsing to Val. Max. sections like 1(ext).

pjheslin committed:
Update Changes.

pjheslin committed:
Fix bug when browsing to "fr" (fragment) at end of file.

pjheslin committed:
Update Changes

pjheslin committed:
Merge branch 'fix_shebang' of https://github.com/luxferresum/diogenes into luxferresum-fix_shebang

pjheslin committed:
Merge branch 'luxferresum-fix_shebang'

Use /usr/bin/env instead of /usr/bin/perl

pjheslin committed:
Fix inadvertent removal of uppercase letters when seeking in TLL.

pjheslin committed:
Remove accents from utf8 Latin lemmas when looking up in TLL and OLD

pjheslin committed:
Tidy-up

pjheslin committed:
Fix for bug caused by OLD bookmarks with upper-case V's.

pjheslin committed:
Add New Testament to list of authors with weird ToC issues.

Faulty or misunderstood ToC data yields error when browsing to
e.g. John 9:16.

pjheslin committed:
First minimally functional effort at chronological searching.

pjheslin committed:
To-Dos

pjheslin committed:
Refactor to simplify reading in chronology file.

pjheslin committed:
Added tlg author date to bibliographical info

pjheslin committed:
Changed Unicode en-dashes to hyphens in tlg chronology file

pjheslin committed:
Whitespace cleanup of tlg chronology file

pjheslin committed:
Minor chronology edits.

pjheslin committed:
Implement chronological searching in particular works.

pjheslin committed:
Implement chronological searching for particular authors.

pjheslin committed:
Print selected list of TLG authors to be searched in chron order.

pjheslin committed:
Implement chronological searching for TLG word-list.

pjheslin committed:
Ensure local dependencies/CPAN for Module::Path.

pjheslin committed:
Add to Changes

pjheslin committed:
To-do

pjheslin committed:
Remove -S option from /usr/bin/env: not yet widely supported.

Previous PR changed from /usr/bin/perl to /usr/bin/env -S to allow
the -w switch for Perl, but current versions of coreutils often do not
support this option.  So "use warnings" instead.

pjheslin committed:
Add Varia and Incertum tlg files to chronology

pjheslin committed:
Make chronological search compatible with uppercase filenames

pjheslin committed:
Fix typo.

pjheslin committed:
Finally fix bug with chron search and uppercase filenames.

pjheslin committed:
Interleave varia and incerta numerically.

pjheslin committed:
Tidy Changes

pjheslin committed:
Fix multiple matching to find repetition of a word in a passage.

The old code presumed that the patterns were distinct and would not
work when you entered the same pattern multiple times.  Now the
repeat_matches flag indicates that all hits in a passage are to be
counted toward the total, rather than counting the number of patterns
that are hit in each passage, which is still the default behaviour.

pjheslin committed:
Add Jiang's updates to the TLG chronology.

pjheslin committed:
Fix scoping bug.

pjheslin committed:
Guard against errors in case TLG files missing in chron. search

pjheslin committed:
Fixes for making wordlists.

pjheslin committed:
Small wordlist fixes.

pjheslin committed:
Add support for new TLL pdf files (but they don't work).

The new batch of PDFs from the TLL do not have bookmarks.

pjheslin committed:
Complied with markdown linting style.

pjheslin committed:
Fix bugs in mk.wordlists

pjheslin committed:
Fixed and linted Readme.md

pjheslin committed:
Get ready for 4.6 release

pjheslin committed:
Makefile fixes for building on Linux

pjheslin committed:
Fix chron search filename paths on Windows.

Volume is already present in tlg_dir.

pjheslin committed:
Tide Changes file.

pjheslin committed:
Better error message when opening PDFs

pjheslin committed:
Auto-select first 0 in passage browser for convenient overwriting.

pjheslin committed:
Add Cicero, Quintilian, Aesop to the list of authors to skip ToC

pjheslin committed:
Updated to-dos

Commit:
Use empty volume to load tlg

pjheslin committed:
Merge pull request #81 from ben-larson/master

Use empty volume to load tlg

pjheslin committed:
Version 4.6 was never released; revert change file.

pjheslin committed:
Add dark mode

pjheslin committed:
Begin migration to Electron v. 20

pjheslin committed:
New security structure using IPC rather than privileged preload

pjheslin committed:
Begin migration of all privileged functionality to main process.

pjheslin committed:
Firstrun page basically working with new framework.

pjheslin committed:
Fix XML export path for new architecture

pjheslin committed:
Fix for File -> Save

pjheslin committed:
Fixed printing to PDF

pjheslin committed:
Changed logic for confirming TLL downloads.

pjheslin committed:
Fixes for find-in-page

pjheslin committed:
Fixes for font selection

pjheslin committed:
Made all new IPC async with invoke and handle, as per latest policy

pjheslin committed:
Display of PDFs now just works (opens in a new window)

pjheslin committed:
Minor housekeeping

pjheslin committed:
Add menu items for top/bottom of page

pjheslin committed:
Make splash screen state semi-persistent

pjheslin committed:
Fixed bug where jumping to context for documentary texts

Went back to the previous document.

pjheslin committed:
Fixed bug where jumping to context wrongly showed an empty sidebar

pjheslin committed:
Fixed bug where morphological inscr. search treated Greek as Latin

pjheslin committed:
Fix bug where part of hyphenated Greek in Latin text got lost

Reported by Mac Pigman in Satyricon 48.8

pjheslin committed:
Change UI for forward/backward in text browser and add jump to loc

pjheslin committed:
To-dos

pjheslin committed:
misc

pjheslin committed:
Fixed bug for words in Pindar with %15 for mid-word apostrophe

pjheslin committed:
misc

pjheslin committed:
New Mac icon with mostly solid background and rounded corners

pjheslin committed:
Fix makefile and Readme to refer to both kinds of Mac app

pjheslin committed:
Remove stray "1" printed on splash page.

pjheslin committed:
misc

pjheslin committed:
Updated Readme instructions on building the Mac icon.

pjheslin committed:
More Readme cleanup.

pjheslin committed:
misc

pjheslin committed:
Experiments with opening PDFs

pjheslin committed:
Get JSON index file from TLL website

pjheslin committed:
New script for TLL bookmarks, using the JSON data from the TLL.

pjheslin committed:
Add new TLL PDF (nebel-nemo) to download script.

pjheslin committed:
Added list of starting pages in all PDFs and simplified logic.

pjheslin committed:
New tll code ready to test

pjheslin committed:
Accommodate new style tll index numbers

pjheslin committed:
Use new TLL code in makefile

pjheslin committed:
New code to open TLL PDFs fully within Electron

pjheslin committed:
Fix positioning of TLL windows

pjheslin committed:
misc

pjheslin committed:
Added support for opening the OLD in the app.

pjheslin committed:
Keyboard navigation for dropdown menus on splash page

pjheslin committed:
Added keybindings to go back and forward in text browser.

The menu item does not work, even though the code follows the docs on
sending keyboard events with modifiers.  It does advertise the
existence of the bindings, however.

pjheslin committed:
misc

pjheslin committed:
Fix parsing TLL index of "do dare", etc.

pjheslin committed:
Fixes for some TLL lemmata

pjheslin committed:
Fixed browse forward/back menu items

pjheslin committed:
Don't abort completely on error parsing TLL's index.json

pjheslin committed:
Use new style TLL filename for nebel-nemo PDF.

pjheslin committed:
Bookmarks in the TLL PDFs have consonant u; index.json has v

pjheslin committed:
Initial fix for accented chars in filter names; needs testing

pjheslin committed:
More filters

pjheslin committed:
Tidying up

pjheslin committed:
Ignore problematic .idt file for Sallust

pjheslin committed:
Disable using .idt file for Scholia in Platonem

pjheslin committed:
By default disable use of ToC from idt file when browsing

It leads to lots bugs with of unreachable text with short works and
it's not at all necessary with today's fast computers.

pjheslin committed:
misc

pjheslin committed:
Rewrite of utf8 filter name handling.

The previous commit did not fix the bugs with utf8 filter names.

pjheslin committed:
Simpler solution to filter name encoding: use numbers always.

Since we are presenting a list, we can select from by number, so we
keep filter names out of the cgi parameters entirely (except for the
unavoidable case where the user is typing in a new name).

pjheslin committed:
misc

pjheslin committed:
Make sure to add listeners after load.

pjheslin committed:
Preliminary work for adding chunking of search output

pjheslin committed:
Implemented paging of large search output.

pjheslin committed:
Forgot one file to add to previous commit.

pjheslin committed:
misc

pjheslin committed:
Fix spinning cursor

pjheslin committed:
Fix footer spacing

pjheslin committed:
Take account of when a chunk ends with the last author for searching

pjheslin committed:
misc

pjheslin committed:
Basic support for searching headwords (Suda, etc.)

pjheslin committed:
Die if tlg chronology file disappears.

pjheslin committed:
Added headword search to the splash page

pjheslin committed:
misc

pjheslin committed:
Add support for Photius' lexicon.

pjheslin committed:
New dates/ranges for many incerta and varia, from Jiang Qian

pjheslin committed:
Experiment with Winge's Morpheus

pjheslin committed:
Updated info about different versions of Morpheus.

pjheslin committed:
Improved handling of special notes in PHI texts in browser and XML

pjheslin committed:
Make singular 1 line a legal value for context.

pjheslin committed:
Disable unnecessary hardware acceleration to avoid crashes.

pjheslin committed:
Show error when using TLL/OLD and path not set.

pjheslin committed:
Layout was breaking at &nbsp; on Windows, so use CSS instead.

pjheslin committed:
Adjust to using utf-8 codepage for Windows via manifest.

pjheslin committed:
Necessary not to slurp file on Windows

pjheslin committed:
Forgot about old_pdf_dir

pjheslin committed:
Avoid error when json file is empty

pjheslin committed:
Add utf-8 manifest for perl.exe

pjheslin committed:
Add tool for adding manifest to exe; from SDK

pjheslin committed:
Apply utf-8 manifest to perl.exe

pjheslin committed:
Fixes for Windows installer build on Linux

pjheslin committed:
Merge branch 'master' of https://github.com/pjheslin/diogenes

pjheslin committed:
Fix for makefile

pjheslin committed:
Merge branch 'master' of https://github.com/pjheslin/diogenes

pjheslin committed:
The correct fix for the Windows splash formatting problem.

pjheslin committed:
Fix for Windows bug with multiple "-- " parameters for location, etc

pjheslin committed:
tidying

pjheslin committed:
Use tll_pdf_dir when downloading, so we can find the PDFs later

pjheslin committed:
Tidy up Readme and changelog

pjheslin committed:
Simplify footer

pjheslin committed:
Fix bug where hit at start of line added extra line of context

Extra line before hit, when using number of lines.

pjheslin committed:
Remove needless stuff from dist dir in Linux app

pjheslin committed:
Guard against accidental make clean

pjheslin committed:
Remove references to the days when only LSJ came from Logeion

pjheslin committed:
Was a bad idea to make lines browsing forward and back off by one

It was supposed to be a clever idea, but it just ended up seeming weird.

pjheslin committed:
Show user error when TLL file not found

pjheslin committed:
fix makfile

pjheslin committed:
minor fixes to makefile

pjheslin committed:
Update website for release of 4.6

pjheslin committed:
Fix for missing JSON.pm in many Perl installs on Mac and Linux.

pjheslin committed:
Upload Windows release separately, from different build machine

pjheslin committed:
Further tidy makefile for releases

pjheslin committed:
Remove spurious / from end of OLD PDF file path.

pjheslin committed:
Fix error when OLD PDF window has been destroyed (win obj persists)

pjheslin committed:
4.6.2 with OLD fixes

pjheslin committed:
Fixes for specious headword hits and associated overflow

pjheslin committed:
Fix accesskey for new Headwords menu item

nickjwhite commented:
Closing as future development should follow https://github.com/pjheslin/diogenes
